src/z3c/pt/tests/test_templates.py

```diff
@@ -198,56 +203,66 @@ def negotiate(self, request):
         orig_i18n = pagetemplate.i18n
         pagetemplate.i18n = i18n
         try:
-            template = pagetemplate.BaseTemplate('<html />')
-            request = 'strings are allowed'
+            template = pagetemplate.BaseTemplate("<html />")
+            request = "strings are allowed"
             template.render(request=request)
             self.assertIs(i18n.request, request)
         finally:
             pagetemplate.i18n = orig_i18n
 
     def test_translate_mv(self):
-        template = pagetemplate.BaseTemplate("""
+        template = pagetemplate.BaseTemplate(
+            """
         <html>
           <body metal:use-macro="m" />
         </html>
-        """)
+        """
+        )
 
         class Macro(object):
             translate = None
+
             def include(self, stream, econtext, *args, **kwargs):
-                self.translate = econtext['translate']
+                self.translate = econtext["translate"]
+
         macro = Macro()
         template.render(m=macro)
 
         self.assertIsNone(macro.translate(pagetemplate.MV))
 
-class TestBaseTemplateFile(unittest.TestCase):
 
+class TestBaseTemplateFile(unittest.TestCase):
     def test_init_with_path(self):
 
         here = os.path.abspath(os.path.dirname(__file__))
 
-        template = pagetemplate.BaseTemplateFile('view.pt', path=here)
+        template = pagetemplate.BaseTemplateFile("view.pt", path=here)
+
+        self.assertEqual(template.filename, os.path.join(here, "view.pt"))
 
-        self.assertEqual(template.filename,
-                         os.path.join(here, 'view.pt'))
 
 class TestBoundPageTemplate(unittest.TestCase):
 
+    # Avoid DeprecationWarning for assertRaisesRegexp on Python 3 while
+    # coping with Python 2 not having the Regex spelling variant
+    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
+                                unittest.TestCase.assertRaisesRegexp)
+
     def test_setattr(self):
         bound = pagetemplate.BoundPageTemplate(None, None)
-        with self.assertRaisesRegexp(AttributeError,
-                                     "Can't set attribute"):
-            setattr(bound, '__self__', 42)
+        with self.assertRaisesRegex(AttributeError, "Can't set attribute"):
+            setattr(bound, "__self__", 42)
```

Comment:
The setattr here is awkward and unnecessary, replace with `bound.__self__ = 42`.

Reply:
Here we go.